target/linux/ath79/generic/base-files/etc/board.d/02_network

```diff
@@ -224,15 +224,16 @@ ath79_setup_interfaces()
 			"0@eth0" "1:wan" "2:lan:3" "3:lan:2"
 		;;
 	engenius,enh202-v1)
-		ucidef_set_interface_wan "eth0"
+		ucidef_set_interface_lan "eth0"
 		ucidef_add_switch "switch0" \
 			"0@eth1" "4:lan:1"
 		;;
 	engenius,ens202ext-v1)
-		ucidef_set_interface_wan "eth1"
+		ucidef_set_interface_lan "eth1"
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:lan"
 		;;
```

Comment:
At this point, you are most likely better off reconfiguring the PHY multiplexer to avoid pushing LAN traffic thru the CPU.

Reply:
sorry this is a new idea for me...how to proceed? any examples?

Reply:
> reconfiguring the PHY multiplexer to avoid pushing LAN traffic thru the CPU

the boards in question with `ucidef_add_switch` have no gigabit capabilities....if that is related...

I don't see any possible performance issue but I would like to know more about what you mean

Reply:
See the AR9341 datasheet on page 47 (Ethernet Subsystem). PHY0 and PHY4 can be XOR'd to the switch or the other MAC respectively. Alternatively, both PHY can be connected to the switch.

Reply:
Yes I'm aware of this, so you mean that I should have both ethernet ports on the internal switch at GMAC1? And is there really an advantage for this (saving CPU usage as you said)?

Reply:
If I understand correctly...

there is a benefit when both ports are used and there is traffic between the ports, otherwise traffic goes via

GMAC0 -- CPU -- GMAC1

where otherwise the internal switch handles that traffic?

Reply:
Yes, this save resources (Not having to allocate continuous memory for the second mac, less CPU interrupts for passing traffic, utilizing the switch offloading features (STP, VLAN).

Reply:
Unless you have a suggestion for something I may be doing wrong, I will open a bug report and link here

in order to have both ports on the internal switch, it would be ports 1 and 5, but 5 is missing from swconfig on ath79
otherwise, configuration would be like this

if I understand correctly, nothing in DTS effects the number of ports you see in swconfig

```
		ucidef_add_switch "switch0" \
			"0@eth0" "1:lan" "5:lan"
```